client/src/partials/budget/new/new_budget.js

```diff
@@ -0,0 +1,357 @@
+    session.autoAdjust = false;
+  }
+
+  function createBudget() {
+    var newBudgets = [];
+    $scope.periods.data.forEach(function (per, index) {
+    newBudgets.push({
+      'account_id' : session.account.id,
+      'period_id'  : per.id,
+      'budget'     : 0.0});
+    });
+
+    connect.post('budget', newBudgets)
+    .then(function () {
+      messenger.success($translate.instant('BUDGET.EDIT.CREATE_OK'));
+      submitAccount(session.account);
+    })
+    .catch(function (err) {
+    messenger.danger($translate.instant('BUDGET.EDIT.CREATE_FAIL'));
+    console.log(err);
+    });
+  }
+
+  function updateBudget() {
+    // Save the budget data for all the periods
+    $http.post('/budget/update', { 
+      params : {
+        budgets   : $scope.budgets.data,
+        accountId : session.account.id
+      }
+    })
+    .success(function () {
+      messenger.success($translate.instant('BUDGET.EDIT.UPDATE_OK'));
+      submitAccount(session.account);
+      session.autoAdjust = false;
+    })
+    .catch(function (err) {
+      messenger.danger($translate.instant('BUDGET.EDIT.UPDATE_FAIL'));
+      console.log(err);
+    });
+  }
+
+  function toggleFreeze(budget) {
+    budget.freeze = !budget.freeze;
+  }
+
+  function startEditing(budget) {
+    budget.editing = true;
+  }
+
+  function endEditing(budget) {
+    budget.editing = false;
+  }
+
+  function recompute() {
+    if (session.autoAdjust) {
+      var totalFrozen = 0.0,  // Total budget that is frozen/editing on the form
+          totalFree   = 0.0,  // Total budget that is NOT frozen/editing on the form
+          numFree     = 0;    // Number of budgets that are not frozen/editing on the form
+
+      // First figure out how much is free
+      $scope.budgets.data.forEach(function (bud) {
+        if (bud.freeze || bud.editing) {
+          totalFrozen += bud.budget;
+        }
+        else {
+          totalFree += bud.budget;
+          numFree++;
+        }
+      });
+
+      totalFree = session.totalBudget - totalFrozen;
+
+      // Redistribute
+      if (numFree > 0) {
+        $scope.budgets.data.forEach(function (bud) {
+          if (!bud.freeze && !bud.editing) {
+            bud.budget = totalFree / numFree;
+          }
+        });
+        session.validTotal = true;
+      }
+    }
+
+    // Double-check the totals
+    var total = $scope.budgets.data.reduce(sumBudget, 0.0);
+
+    if (isNaN(session.totalBudget) || session.totalBudget === null ||
+      isNaN(total) || total === null) {
+      // Make sure we have real numbers
+      session.validTotal = false;
+    } else {
+      session.validTotal = precision.round(total, 6) === precision.round(session.totalBudget, 6);
+    }
+  }
+
+  function sumBudget(a, b) {
+    return a + b.budget;
+  }
+
+  function loadFiscalYears(models) {
+    angular.extend($scope, models);
+    // Default to the last fiscal year
+    session.fiscal_year = $scope.fiscal_years.data[$scope.fiscal_years.data.length - 1];
```

Comment:
This may become a problem later.  Is there anyway to default to the current fiscal year, _and then_ the last one, if we do not have a fiscal year?